examples/tv-app/android/java/MediaPlaybackManager.cpp

```diff
@@ -92,40 +92,160 @@ uint64_t MediaPlaybackManager::HandleGetSeekRangeEnd()
+
+    ChipLogProgress(Zcl, "MediaPlaybackManager::HandleGetActiveAudioTrack");
+    VerifyOrExit(mMediaPlaybackManagerObject != nullptr, err = CHIP_ERROR_INCORRECT_STATE);
+    VerifyOrExit(mGetActiveTrackMethod != nullptr, err = CHIP_ERROR_INCORRECT_STATE);
+    VerifyOrExit(env != NULL, err = CHIP_JNI_ERROR_NO_ENV);
+
+    env->ExceptionClear();
+    trackObj = env->CallObjectMethod(mMediaPlaybackManagerObject, mGetActiveTrackMethod, static_cast<jboolean>(audio));
+    if (env->ExceptionCheck())
+    {
+        ChipLogError(AppServer, "Java exception in MediaPlaybackManager::HandleGetActiveAudioTrack");
+        env->ExceptionDescribe();
+        env->ExceptionClear();
+        goto exit;
+    }
+
+    if (trackObj != nullptr)
+    {
+        jclass trackClass   = env->GetObjectClass(trackObj);
+        jfieldID getIdField = env->GetFieldID(trackClass, "id", "Ljava/lang/String;");
+        jstring jid         = static_cast<jstring>(env->GetObjectField(trackObj, getIdField));
+        JniUtfString id(env, jid);
+        if (jid != nullptr)
+        {
+            response.id = id.charSpan();
+        }
+
+        jfieldID getLanguageCodeField = env->GetFieldID(trackClass, "languageCode", "Ljava/lang/String;");
+        jstring jlanguagecode         = static_cast<jstring>(env->GetObjectField(trackObj, getLanguageCodeField));
+        JniUtfString languageCode(env, jlanguagecode);
+        if (jlanguagecode != nullptr)
+        {
+            trackAttributes.languageCode = languageCode.charSpan();
+        }
+
+        jfieldID getDisplayNameField = env->GetFieldID(trackClass, "displayName", "Ljava/lang/String;");
+        jstring jdisplayname         = static_cast<jstring>(env->GetObjectField(trackObj, getDisplayNameField));
+        JniUtfString displayName(env, jdisplayname);
+        if (jdisplayname != nullptr)
+        {
+            trackAttributes.displayName = Optional<chip::app::DataModel::Nullable<chip::Span<const char>>>(
+                Nullable<chip::Span<const char>>(displayName.charSpan()));
+        }
+    }
+    else
+    {
+        err = aEncoder.EncodeNull();
+    }
+
+exit:
+    if (err != CHIP_NO_ERROR)
+    {
+        ChipLogError(Zcl, "MediaPlaybackManager::HandleGetActiveAudioTrack status error: %s", err.AsString());
+    }
+
+    return aEncoder.Encode(response);
 }
 
 CHIP_ERROR MediaPlaybackManager::HandleGetAvailableAudioTracks(AttributeValueEncoder & aEncoder)
 {
-    std::vector<TrackType> mAvailableAudioTracks;
-    // TODO: Insert code here
-    return aEncoder.EncodeList([mAvailableAudioTracks](const auto & encoder) -> CHIP_ERROR {
-        for (auto const & audioTrack : mAvailableAudioTracks)
+    return HandleGetAvailableTracks(true, aEncoder);
+}
+
+CHIP_ERROR MediaPlaybackManager::HandleGetAvailableTracks(bool audio, AttributeValueEncoder & aEncoder)
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+    JNIEnv * env   = JniReferences::GetInstance().GetEnvForCurrentThread();
+
```

Comment:
Could you put the below JniLocalReferenceManager in the beginning of this HandleRecordProgram callback function so that we can automatically release local reference , when it is out of scope.
JNIEnv * env = JniReferences::GetInstance().GetEnvForCurrentThread();
VerifyOrReturn(env != nullptr, ChipLogError(Controller, "Could not get JNIEnv for current thread"));
JniLocalReferenceManager manager(env);

Reply:
fixed - thanks!